src/broker/cas_cgw.c

```diff
@@ -2438,3 +2456,273 @@ cgw_utype_to_string (int type)
+	}
+
+      if (cols_list[i - 1] == NULL || strcmp (cols_list[i - 1], "") == 0)
+	{
+	  is_rewrite_error = true;
+	  goto REWRITE_ERROR;
+	}
+
+      cgw_strcat_with_realloc (&new_inline_view, col_name);
+      cgw_strcat_with_realloc (&new_inline_view, " AS ");
+      cgw_strcat_with_realloc (&new_inline_view, cols_list[i - 1]);
+
+      if (i <= num_cols - 1)
+	{
+	  cgw_strcat_with_realloc (&new_inline_view, ", ");
+	}
+      else
+	{
+	  cgw_strcat_with_realloc (&new_inline_view, " FROM ");
+	  cgw_strcat_with_realloc (&new_inline_view, inline_view);
+	  cgw_strcat_with_realloc (&new_inline_view, ")");
+	}
+    }
+
+  rewrite_query = (char *) MALLOC (1);
+  memset (rewrite_query, 0x0, 1);
+  cgw_strcat_with_realloc (&rewrite_query, select_from);
+  cgw_strcat_with_realloc (&rewrite_query, " ");
+  cgw_strcat_with_realloc (&rewrite_query, new_inline_view);
+  cgw_strcat_with_realloc (&rewrite_query, " ");
+  cgw_strcat_with_realloc (&rewrite_query, where);
+
+REWRITE_ERROR:
+  SQLFreeHandle (SQL_HANDLE_STMT, hstmt);
+
+  if (select_from != NULL)
+    {
+      FREE_MEM (select_from);
+    }
+
+  if (inline_view != NULL)
+    {
+      FREE_MEM (inline_view);
+    }
+
+  if (cublink != NULL)
+    {
+      FREE_MEM (cublink);
+    }
+
+  if (cols_type != NULL)
+    {
+      FREE_MEM (cols_type);
+    }
+
+  if (cols_list != NULL)
+    {
+      cgw_free_string_array (cols_list);
+    }
+
+  if (where != NULL)
+    {
+      FREE_MEM (where);
+    }
+
+  if (rewrite_query != NULL && is_rewrite_error == TRUE)
+    {
+      FREE_MEM (rewrite_query);
+      rewrite_query = NULL;
+    }
+
+  return rewrite_query;
+}
+
+static void
+cgw_eliminate (char *str, char ch)
+{
+  size_t len = strlen (str) + 1;
+  for (; *str != '\0'; str++, len--)
+    {
+      if (*str == ch || *str == ' ')
+	{
+	  strncpy (str, str + 1, len);
+	  str--;
+	}
+    }
+}
+
+static void
+cgw_free_string_array (char **array)
+{
+  int i;
+
+  for (i = 0; array[i] != NULL; i++)
+    {
+      FREE_MEM (array[i]);
+    }
+  FREE_MEM (array);
+}
+
+static char **
+cgw_split_string (const char *str, const char *delim)
+{
+  char *t, *o;
+  char *v;
+  char **r = NULL;
+  int count = 1;
+
+  if (str == NULL)
+    {
+      return NULL;
+    }
+
+  o = strdup (str);
+  if (o == NULL)
+    {
+      return NULL;
+    }
+
+  for (t = o;; t = NULL)
+    {
+      v = strtok (t, delim);
+      if (v == NULL)
+	{
+	  break;
+	}
+      char **const realloc_r = (char **) realloc (r, sizeof (char *) * (count + 1));
+      if (realloc_r == NULL)
+	{
+	  FREE_MEM (o);
+	  return NULL;
+	}
+      else
+	{
+	  r = realloc_r;
+	}
+      r[count - 1] = strdup (v);
+      r[count] = NULL;
+      count++;
+    }
+
+  FREE_MEM (o);
+  return r;
+}
+
+static void
+cgw_strcat_with_realloc (char **src, const char *tail)
+{
+  char *cp = *src;
+
+  *src = (char *) realloc (*src, strlen (*src) + strlen (tail) + 1);
+  if (*src == NULL)
+    {
+      FREE_MEM (cp);
+    }
+  cp = *src;
+  strcat (cp, tail);
```

Comment:
if "*src" is NULL, strcat is failed and will araise segment violation.

Reply:
removed cgw_strcat_with_realloc()